### PR TITLE
[Sprint: 51][1.2 GA] XD-3115 Support Reactor Processors that have an inheritance hierarchy

### PR DESCRIPTION
**Pull request by mbogoevici**
- Navigate the class hierarchy for finding the parameterized Processor interface
- Ensure that the Processor interface is identified correctly for extracting type arguments
- Correct behaviour if a raw type processor is used
- Ensure that Message instances are passed directly if and only if the Processor defines a Message type argument

### CODE REVIEW

File: spring-xd-reactor/src/main/java/org/springframework/xd/reactor/BroadcasterMessageHandler.java [hunk 1]
```diff
@@ -90,7 +88,7 @@ public BroadcasterMessageHandler(Processor processor) {
         this.reactorProcessor = processor;
         Environment.initializeIfEmpty(); // This by default uses SynchronousDispatcher
 
-        this.inputType = ReactorReflectionUtils.extractGeneric(processor);
+        this.inputType = ReactorReflectionUtils.extractInputType(processor);
 
         //Stream with a RingBufferProcessor
         this.stream = RingBufferProcessor.share("xd-reactor", 8192); //todo expose the backlog size in module conf
```
Comment (markpollack) [inline spring-xd-reactor/src/main/java/org/springframework/xd/reactor/BroadcasterMessageHandler.java]: Will add setter for bufferSize on merge.

Comment (markpollack) [inline spring-xd-reactor/src/main/java/org/springframework/xd/reactor/BroadcasterMessageHandler.java]: Oops, forget this, looking at wrong PR

**Comment by mbogoevici** [comment]
Rebased after https://github.com/spring-projects/spring-xd/commit/87e6d3f7adf7c1357f6d962c4e675a7644d965be

**Comment by markpollack** [comment]
tidy up formatting and merged as 60280d21b81e79d7caf350ffc63c69d551a04e6c

**Pull request closed without merging**